src/mono/wasm/runtime/memory.ts

```diff
@@ -60,13 +61,23 @@ export function setI16(offset: _MemOffset, value: number): void {
     Module.HEAP16[<any>offset >>> 1] = value;
 }
 
-export function setI32 (offset: _MemOffset, value: _NumberOrPointer) : void {
+export function setI32(offset: _MemOffset, value: _NumberOrPointer): void {
     Module.HEAP32[<any>offset >>> 2] = <number><any>value;
 }
 
-// NOTE: Accepts a number, not a BigInt, so values over Number.MAX_SAFE_INTEGER will be corrupted
+/**
+ * NOTE: Accepts a Number, not a BigInt. 
+ * This means values over Number.MAX_SAFE_INTEGER will be converted to floating point representation (double).
+ * Number(BigInt(Number.MAX_SAFE_INTEGER)+BigInt(2)) == 9007199254740992
+ * Number(BigInt(Number.MAX_SAFE_INTEGER)+BigInt(3)) == 9007199254740994 (notice the +1 error)
+ * See also setI64Big()
+ */
 export function setI64(offset: _MemOffset, value: number): void {
-    Module.setValue(<VoidPtr><any>offset, value, "i64");
+    HEAPI64[<any>offset >>> 3] = BigInt(<number>value);
```

Comment:
@kg noted that maybe bitwise math would be faster than the allocation. we should explore that.

Reply:
done